bindings/java/src/org/sleuthkit/datamodel/Blackboard.java

```diff
@@ -123,7 +146,93 @@ public void postArtifacts(Collection<BlackboardArtifact> artifacts, String modul
 			throw new BlackboardException("Failed to get or add artifact type", ex);
 		}
 	}
-
+	
+	/**
+	 * Adds new analysis result artifact.
+	 *
+	 * @param artifactType      Type of analysis result artifact to create.
+	 * @param obj_id            Object id of parent.
+	 * @param datasource_obj_id Data source object id.
+	 * @param score	            Score associated with this analysis result.
+	 * @param conclusion        Conclusion of the analysis, may be null or an
+	 *                          empty string.
+	 * @param configuration     Configuration association with this analysis,
+	 *                          may be null or an empty string.
+	 * @param justification     Justification, may be null or an empty string.
+	 * @param attributesList    Attributes to be attached to this analysis
+	 *                          result artifact.
+	 * @param transaction       DB transaction to use.
+	 *
+	 * @return Analysis result created.
+	 *
+	 * @throws BlackboardException exception thrown if a critical error occurs
+	 *                             within TSK core
+	 */
+	public AnalysisResult newAnalysisResult(BlackboardArtifact.Type artifactType, long obj_id, long datasource_obj_id, Score score, String conclusion, String configuration, String justification, Collection<BlackboardAttribute> attributesList, CaseDbTransaction transaction) throws BlackboardException {
+		
+		try {
+			AnalysisResult analysisResult =  caseDb.newAnalysisResult(artifactType, obj_id, datasource_obj_id, score, conclusion, configuration, justification, transaction.getConnection());
+			
+			// add the given attributes
+			addBlackboardAttributes(attributesList, artifactType.getTypeID(), analysisResult.getArtifactID(), transaction);
+			
+			return analysisResult;
+		}  catch (TskCoreException ex) {
+			throw new BlackboardException("Failed to add analysis result.", ex);
+		}
+	}
+	
+	/**
+	 * Get all analysis results of a given type. 
+	 *
+	 * @param artifactTypeID Type of results to get.
+	 * @param dataSourceObjId Data source to look under
+	 *
+	 * @return list of analysis results.
+	 *
+	 * @throws TskCoreException exception thrown if a critical error occurs
+	 *                          within TSK core
+	 */
+	public List<AnalysisResult> getAnalysisResults(int artifactTypeID, long dataSourceObjId) throws TskCoreException {
```

Comment:
We should check the category of the artifact type passed in by the caller and immediately throw an exception if the type is not in the analysis result category.